README.md

```diff
@@ -20,16 +20,125 @@
+
+Once the `voting_duration` has passed, votes are closed, and anyone can call `finalize_proposal` (this time directly on the space contract as no authentication is required): this will finalize the proposal, count the votes for/against/abstain, and call the `executor`. 
+
+Note that each DAO will have at least one space, however a DAO might choose to have multiple spaces if they want to create different 'categories' of proposal each with different settings.
+
+### Voting Strategies
+
+Voting strategies are the contracts used to determine the voting power of a user. Voting strategies can be created in a permissionless way however to use one, one must whitelist the strategy contract in the relevant space contract for the DAO. The most common example is using the ERC-20 token balance of a user to determine his voting power. But we could imagine other voting strategies: owning a specific NFT, owning NFT of collection X and another NFT of collection Y, having participated in protocol xyz... the possibilities are endless! [2] We provide the [single_slot_proof strategy](contracts/starknet/strategies/single_slot_proof.cairo) which allows classic ERC20 and ERC721 balances on L1 (thanks to [Fossil](#Fossil-Storage-Verifier)) to be used as voting power, but feel free to create your own strategies! We hope that the flexibility of the system will unlock a new era of programmable on-chain governance. The interface of a strategy can be found [here](contracts/starknet/strategies/interface.cairo). 
+
+### Fossil Storage Verifier
+
+The backbone of the voting strategies is the Fossil module built by the awesome Oiler team. This module allows any part of the Ethereum mainnet state to be trustlessly verified on StarkNet. Verification of Ethereum state information is achieved by submitting a proof of the state to StarkNet and then verifying that proof. Once this state information has been proved, we can then calculate voting power as an arbitrary function of the information. For more information on Fossil, refer to their [Github](https://github.com/OilerNetwork/fossil)
+
+### Authenticators
+
+Authenticators are the contracts in charge of authenticating users. All authenticators have a function titled `execute` that takes the following arguments: 
+- `to`: The address of the space contract the user wants to interact with. 
+- `function selector`: The function selector for the function inside the space contract that the user wants to invoke (vote or propose).
+- `calldata`: An array of parameters that are required by the function specified by `function selector`. 
+
+Beyond this, each authenticator implements different logic depending on the type of authentication that is being done. This repository provides three useful authenticators:
+- [Ethereum Signature Authenticator](contracts/starknet/authenticator/ethereum.cairo): Will authenticate a user based on a message signed by Ethereum private keys.

```

Comment:
@bonustrack I'm tagging you here regarding the comment you made saying we should use something like `ECDSA` rather than Ethereum...
The problem with that is the curve used is `secp256k1` so I think it should be `ECDSA secp256k1` which is pretty long...
We could maybe go for `EVM Signature Authenticator`? wdyt?

Reply:
@pscott Never mind we can go for Ethereum signature, I feel EVM would be more confusing

Reply:
Yeah. Agreed Ethereum is not THE ideal solution, but I feel like it's currently the best...

Reply:
shall we change the cairo file to `ethereum_signature.cairo` then? And while we are at it, probably better to call L1 transaction authenticator `Ethereum Transaction Authenticator` and cairo file `ethereum_transaction.cairo`

Reply:
Ill sort of use L1 and Ethereum synonomously

Reply:
Yeah I think we should try to use `Ethereum` as much as possible, and keep `l1` really only for technical stuff / places where it's useful to have "short" naming?

Reply:
agreed

Reply:
same with Starknet and L2